### PR TITLE
cmd/bosun: 2x API routes: Metadata for all metrics, metrics per tagk

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

File: cmd/bosun/web/search.go [hunk 1]
```diff
@@ -63,6 +63,24 @@ func MetricsByTagPair(t miniprofiler.Timer, w http.ResponseWriter, r *http.Reque
 	return schedule.Search.MetricsByTagPair(tagk, tagv)
 }
 
+func MetricsByTagKey(t miniprofiler.Timer, w http.ResponseWriter, r *http.Request) (interface{}, error) {
```
Comment (captncraig) [inline cmd/bosun/web/search.go]: Perhaps a comment explaining what it returns with an example. map[string][]string could be anything

File: cmd/bosun/web/web.go [hunk 2]
```diff
@@ -358,12 +360,58 @@ func GetMetadata(t miniprofiler.Timer, w http.ResponseWriter, r *http.Request) (
 	return schedule.GetMetadata(r.FormValue("metric"), tags)
 }
 
+type MetricMetaTagKeys struct {
+	*database.MetricMetadata
```
Comment (captncraig) [inline cmd/bosun/web/web.go]: This type should be moved into models now. But that can wait until later

File: cmd/bosun/web/web.go [hunk 3]
```diff
@@ -358,12 +360,58 @@ func GetMetadata(t miniprofiler.Timer, w http.ResponseWriter, r *http.Request) (
...
+	TagKeys []string
+}
+
 func MetadataMetrics(t miniprofiler.Timer, w http.ResponseWriter, r *http.Request) (interface{}, error) {
```
Comment (captncraig) [inline cmd/bosun/web/web.go]: I'm gonna insist on a better name here. MetadataMetrics could mean anything

Comment (kylebrandt) [inline cmd/bosun/web/web.go]: It Matches the underlying schedule name:

677f012e cmd/bosun/sched/sched.go (Craig Peterson 2015-09-25 14:05:28 -0600 161) func (s *Schedule) MetadataMetrics(metric string) (*database.MetricMetadata, error) {

;-)

**Comment by captncraig** [comment]
lgtm

**Pull request merged**